nengo_viz/server.py

```diff
@@ -45,9 +45,18 @@ def swi_shutdown(self, *path):
     def ws_viz_component(self, client, uid):
         """Handles ws://host:port/viz_component with a websocket"""
         # figure out what component is being connected to
+
+        viz_sim = Server.viz_sim
+
+        component = viz_sim.uids[uid]
         try:
             while True:
-                component = self.viz_sim.uids[uid]
+                if viz_sim.finished:
+                    component.finish()
+                    return
```

Comment:
I'd use `break` instead of `return` here just because I think it's a bit clearer, but no biggie either way.